otlp/src/otlp/logs/component.cpp

```
@@ -21,6 +22,18 @@ USERVER_NAMESPACE_BEGIN
 
 namespace otlp {
 
+namespace {
+SinkType GetSinkTypeFromString(const std::string& destination) {
+  if (destination == "both") {
+    return SinkType::kBoth;
+  }
+  if (destination == "default") {
+    return SinkType::kDefault;
+  }
+  return SinkType::kOtlp;
```

you need an explicit check, otherwise a typo would silently default to otlp

---

does enum in schema not mean implicit check on yaml config level?

---

explicit is better than implicit here